app/src/lib/dispatcher/validated-repository-path.ts

```diff
@@ -1,15 +1,12 @@
-import * as Path from 'path'
-
-import { getGitDir } from '../git'
+import { getTopLevelWorkingDirectory } from '../git'
 
 /**
  * Get the path to the parent of the .git directory or null if the path isn't a
  * valid repository.
  */
 export async function validatedRepositoryPath(path: string): Promise<string | null> {
   try {
-    const gitDir = await getGitDir(path)
-    return gitDir ? Path.dirname(gitDir) : null
+    return await getTopLevelWorkingDirectory(path)
```

Comment:
In this method I just swapped out the method call but the await is there so that we can catch any errors emanating from the `rev-parse` call. Seeing how we already deal with the 128 exit code case in `getTopLevelWorkingDirectory` I could certainly see us not swallowing all errors here though. I guess that would suffer from the same risk that @iAmWillShepherd talked about earlier with invalid paths throwing.

I'm tentatively 👍 with changing the behavior but I'd like some more time to think about edge-cases so I'm proposing we merge this with the catch and I'll have a closer look and follow up in a PR. This method is starting to look pretty useless if we remove the catch so we might be able to drop it and just call `getTopLevelWorkingDirectory` directly